packages/ra-core/src/controller/list/useInfiniteListController.spec.tsx

```diff
@@ -45,6 +46,93 @@ describe('useInfiniteListController', () => {
         debounce: 200,
     };
 
+    const dataProvider = testDataProvider({
+        getList: jest.fn().mockImplementation((_resource, params) =>
+            Promise.resolve({
+                data: [{ id: 0 }, { id: 1 }].slice(
+                    0,
+                    params.pagination.perPage
+                ),
+                total: 2,
+            })
+        ),
+    });
+
+    describe('onSelectAll', () => {
+        it('should select all items if no items are selected', async () => {
+            const children = jest.fn().mockReturnValue(<span>children</span>);
+            render(<Basic dataProvider={dataProvider} children={children} />);
+            act(() => {
+                children.mock.calls.at(-1)[0].onSelectAll();
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        selectedIds: [0, 1],
+                    })
+                );
+            });
+        });
+        it('should select all items if some items are selected', async () => {
+            const children = jest.fn().mockReturnValue(<span>children</span>);
+            render(<Basic dataProvider={dataProvider} children={children} />);
+            act(() => {
+                children.mock.calls.at(-1)[0].onSelect([0]);
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        selectedIds: [0],
+                    })
+                );
+            });
+            act(() => {
+                children.mock.calls.at(-1)[0].onSelectAll();
+            });
+            await waitFor(() => {
+                expect(children).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        selectedIds: [0, 1],
+                    })
+                );
+            });
+        });
+        it('should select the maximum items possible until we reached the limit', async () => {
+            const getList = jest.fn().mockImplementation((_resource, params) =>
```

Comment:
Why do you need to reimplement it? It's the same data provider as the other tests.